modules/apps/foundation/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/ScopeKey.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.configuration.settings.internal;
+
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.Validator;
+
+/**
+ * @author Drew Brokke
+ */
+public class ScopeKey {
+
+	public ScopeKey(
+		Class<?> objectClass, ExtendedObjectClassDefinition.Scope scope,
+		String scopePrimKey) {
+
+		if (objectClass == null) {
+			throw new IllegalArgumentException(
+				"A scoped PID must correspond to an existing configuration " +
+					"class");
+		}
+
+		if (scope == null) {
+			throw new IllegalArgumentException(
+				"A scoped PID must contain one of the following scope " +
+				"separators: __COMPANY__ , __GROUP__ , __PORTLET_INSTANCE__");
+		}
```

Comment:
I think the message should be worded to something like "A ScopeKey must correspond to one of the following scope options: company, group, or portlet instance." since the scope is an instance of ExtendedObjectClassDefinition.Scope, and we're not parsing the scoped pid here.

Reply:
That makes sense. Perhaps I won't mention anything about a scoped pid here in any of the exception messages, especially if we find we're able to not use them at some point.